### PR TITLE
Added Random Forest Algorithm Implementation

### PR DESCRIPTION
**Pull request by manishkumar00208**
### Describe your change:
Implementation of Random Forest Algorithm (from scratch) using Python - A Machine Learning Model/Algorithm.

* [✔️] Add an algorithm?
Implemented Random Forest Classifier for Machine Learning using Python (No Libraries are used).
* [x] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [✔️] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [✔️] This pull request is all my own work -- I have not plagiarized.
* [✔️] I know that pull requests will not be merged if they fail the automated tests.
* [✔️] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [✔️] All new Python files are placed inside an existing directory.
* [✔️] All filenames are in all lowercase characters with no spaces or dashes.
* [✔️] All functions and variable names follow Python naming conventions.
* [✔️] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [✔️] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [✔️] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.

Screenshot:
image

Note:
Directly run the program for the execution. It provides the robust functionality of RandomForest Classifier. This program will build 10 decision tree models to use as a ensemble in RandomForest classifer, and is tested with all 3 datasets provided. The accuracy of the classifier is almost same (sometimes a bit high, sometimes a bit low) as we obtain from the Weka toolkit.

### CODE REVIEW

File: machine_learning/Random Forest/RandomForest.py [hunk 1]
```diff
@@ -0,0 +1,520 @@
+"""
```
Comment (cclauss) [inline machine_learning/Random Forest/RandomForest.py:1]: Please rename the file path to `machine_learning/random_forest/random_forest.py`

File: machine_learning/Random Forest/RandomForest.py [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+"""
+list of libraries imported for the program to work.
+"""
+import math
+import numpy as np
+import csv
+from collections import defaultdict
+import operator
+import random
```
Comment (cclauss) [inline machine_learning/Random Forest/RandomForest.py:13]: Please use isort sort order.
```suggestion
import csv
import operator
import math
import random
from collections import defaultdict
from dataclasses import dataclass

import numpy as np
```

File: machine_learning/Random Forest/RandomForest.py [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+I have called this class in RandomForest Class to build multiple tree models
+and saved the objects and models in RandomForest class to use for classification
+
+Building a Decision Tree and testing it on  training data.
+I've separated the Decision Tree classifier and related function,
+so that I can even test it independently.
+"""
+
+
+class DecisionTree(object):
```
Comment (cclauss) [inline machine_learning/Random Forest/RandomForest.py:26]: Please make this a https://docs.python.org/3/library/dataclasses.html
```suggestion
@dataclass
class DecisionTree:
```

File: machine_learning/Random Forest/RandomForest.py [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+    class attributes; classIndex define the target class index in the data. In
+    our case target class index is always the last index of the data. training
+    sampels is the training data provided to Decision Tree Class every time
+
+    We're getting random bootstrap sample data with replacement every time
+    for each decision tree model built:
+    """
+
+    classIndex = 0
+    trainingSamples = []
```
Comment (cclauss) [inline machine_learning/Random Forest/RandomForest.py:37]: This is Python code....
```suggestion
    class_index = 0
    training_samples = []
```

File: machine_learning/Random Forest/RandomForest.py [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+
+    """
+    init function to initialize the classIndex and trainingSamples
+    """
+
+    def __init__(self, classIndex, trainingSamples):
+        self.classIndex = classIndex
+        self.trainingSamples = trainingSamples
```
Comment (cclauss) [inline machine_learning/Random Forest/RandomForest.py:45]: Dataclasses make this easier...  All methods will need type hints and doctests.
```suggestion
    class_index: int
    training_samples: list[]
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Please read CONTRIBUTING.md to understand naming, type hint, and doctest requirements.

**Comment by cclauss** [comment]
Closing tests_are_failing PRs to prepare for Hacktoberfest 2024
